app/docker/views/configs/create/createConfigController.js

```diff
@@ -53,6 +62,21 @@ class CreateConfigController {
     }
   }
 
+  async uiCanExit() {
+    if (this.formValues.displayCodeEditor && this.formValues.ConfigContent && this.state.isEditorDirty) {
+      return this.ModalService.confirmAsync({
```

Comment:
The confirmAsync is duplicated many times. It would be nice if we could make it reusable by moving it into ModalService.